dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/cloudera/manager/ClouderaManagerHandle.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2022-2024 Google LLC
+ * Copyright 2013-2021 CompilerWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.application.dumper.connector.cloudera.manager;
+
+import com.google.common.collect.ImmutableList;
+import com.google.edwmigration.dumper.application.dumper.handle.Handle;
+import java.io.IOException;
+import java.net.URI;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import javax.annotation.concurrent.ThreadSafe;
+import org.apache.http.impl.client.CloseableHttpClient;
+
+@ThreadSafe
+public class ClouderaManagerHandle implements Handle {
+  private final URI apiURI;
+  private final CloseableHttpClient httpClient;
+
+  private List<ClouderaClusterDTO> clusters;
+
+  public ClouderaManagerHandle(URI apiURI, CloseableHttpClient httpClient) {
+    this.apiURI = apiURI;
+    this.httpClient = httpClient;
+  }
+
+  public URI getApiURI() {
+    return apiURI;
+  }
+
+  public URI getBaseURI() {
+    return apiURI.resolve("/");
+  }
+
+  public CloseableHttpClient getHttpClient() {
+    return httpClient;
+  }
+
+  @CheckForNull
+  public synchronized List<ClouderaClusterDTO> getClusters() {
+    return clusters;
+  }
+
+  public synchronized void initClusters(List<ClouderaClusterDTO> clusters) {
+    if (this.clusters != null) {
+      throw new IllegalStateException("The cluster already initialized!");
+    }
+    this.clusters = ImmutableList.copyOf(clusters);
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (httpClient != null) {
+      try {
+        httpClient.close();
+      } catch (IOException ignore) {
+
```

Comment:
Could you explain what are the exceptions that are caught here in practive? Is it related to some faulty code outside our control?

Reply:
In practice I do not expect any exceptions here, however the interface declare that it _might_ be an exception and the idea here is to keep plain program execution flow running because we have nothing to do with the exception.

In general, this pattern is similar to try/with-resource implementation The spec: https://docs.oracle.com/javase/specs/jls/se16/html/jls-14.html#jls-14.20.3

Reply:
The comment with the explanation added.